src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_ForEachStatement.cs

```diff
@@ -506,8 +501,8 @@ private BoundStatement RewriteStringForEachStatement(BoundForEachStatement node)
             if (deconstruction == null)
             {
                 // V v = /* expression */
-                iterationVarDecl = MakeLocalDeclaration(forEachSyntax, iterationVar, iterationVarValue);
-                iterationVariables = ImmutableArray.Create(iterationVar);
+                Debug.Assert(iterationVariables.Length == 1);
```

Comment:
Added tests - with a tuple/deconstruction and without

---
In reply to: [95267564](https://github.com/dotnet/roslyn/pull/16318#discussion_r95267564) [](ancestors = 95267564,95262886)